src/cpg_flow/workflow.py

```diff
@@ -163,6 +163,24 @@ def run_workflow(
     return wfl
 
 
+def _compute_shadow(graph: nx.DiGraph, shadow_casters: set[str]) -> set[str]:
+    """Compute the 'shadow' of a set of nodes on a directed graph.
+
+    Shadowed nodes are those that are only not reachable from any root of the
+    graph without passing through a shadow caster node."""
```

Comment:
Surprised this wasn't a google pystyle [docstring](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) 👀

Reply:
Hah. Good point. Although it's quite common for that to be enforced only on public APIs, where it matters more. The linter only complains if you look like you're attempting do document args or return types, and haven't done it properly. LMK if you want me to change it.